### PR TITLE
fixes for compiler warnings

### PR DESCRIPTION
**Pull request by MichaelBrim**
### Description

Fixes most of the compiler warnings generated by XL on Summitdev

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### CODE REVIEW

**Comment by MichaelBrim** [comment]
need to skip meta/src files for checkpatch. is there any way to do that automatically - like an ignore prefix path?

File: server/src/unifycr_sock.c [hunk 1]
```diff
@@ -101,7 +101,7 @@ int sock_init_server(void)
     poll_set[0].events = POLLIN | POLLHUP;
     poll_set[0].revents = 0;
     num_fds++;
-    printf("completed sock init server\n");
+    LOG(LOG_DBG, "completed sock init server\n");
```
Comment (adammoody) [inline server/src/unifycr_sock.c]: The LOG macro will add the \n, so let's drop that too from these new LOG lines.

**Comment by adammoody** [comment]
Thanks, @MichaelBrim

File: server/src/unifycr_sock.c (hunk 1 above)
Comment (MichaelBrim) [reply server/src/unifycr_sock.c]: will do

**Pull request merged**